internal/core/metadata/rest_deviceprofile_test.go

```diff
@@ -692,10 +742,17 @@ func TestUpdateDeviceProfile(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			Configuration = &ConfigurationStruct{Writable: WritableInfo{EnableValueDescriptorManagement: tt.enableValueDescriptorManagement}, Service: config.ServiceInfo{MaxResultCount: 1}}
+			Configuration = &ConfigurationStruct{
+				Writable: WritableInfo{EnableValueDescriptorManagement: tt.enableValueDescriptorManagement},
+				Service:  config.ServiceInfo{MaxResultCount: 1}}
 			vdc = MockValueDescriptorClient{}
 			rr := httptest.NewRecorder()
-			restUpdateDeviceProfile(rr, tt.request, logger.NewMockClient(), tt.dbMock)
+			restUpdateDeviceProfile(
+				rr,
+				tt.request,
+				logger.NewMockClient(),
+				tt.dbMock,
+				errorconcept.NewErrorHandler(logger.NewMockClient()))
```

Comment:
Called out earlier if `errorconcept.Handler` should be a mock.

If you didn't change the existing code to replace the `NewErrorHandler` call with a mock, I'd comment that you're using two different logger instances in the same test where you should only be using one.